### PR TITLE
more prerefactor for roles

### PR DESCRIPTION
**Pull request by snopoke**
## Summary
Minor changes.

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Automated test coverage
Existing tests

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

**Review by orangejenny — APPROVED** [review]

File: corehq/apps/users/models.py [hunk 1]
```diff
@@ -427,22 +424,23 @@ def init_domain_with_presets(cls, domain):
...
     @classmethod
     def get_preset_role_id(cls, name):
         return UserRolePresets.get_preset_role_id(name)
 
     @classmethod
     def preset_and_domain_role_names(cls, domain_name):
-        return cls.preset_permissions_names().union(set([role.name for role in cls.by_domain(domain_name)]))
+        presets = set(UserRolePresets.NAME_ID_MAP.keys())
+        custom = set([role.name for role in cls.by_domain(domain_name)])
+        return presets | custom
```
Comment (esoergel) [inline corehq/apps/users/models.py:438]: :+1:

File: corehq/apps/users/models.py [hunk 2]
```diff
@@ -412,10 +411,11 @@ def archive_custom_roles_for_domain(cls, domain):
 
     @classmethod
     def unarchive_roles_for_domain(cls, domain):
-        archived_roles = cls.by_domain(domain, is_archived=True)
+        archived_roles = cls.by_domain(domain, include_archived=True)
         for role in archived_roles:
```
Comment (esoergel) [inline corehq/apps/users/models.py:415]: nit:
```suggestion
        all_roles = cls.by_domain(domain, include_archived=True)
        for role in all_roles:
```

File: corehq/apps/users/migrations/0024_add_login_as_all_users.py [hunk 3]
```diff
@@ -8,12 +9,7 @@
 
 @skip_on_fresh_install
 def migrate_login_as_all_users_permission(apps, schema_editor):
-    roles = UserRole.view(
-        'users/roles_by_domain',
-        include_docs=False,
-        reduce=False
-    ).all()
-    for role_doc in iter_docs(UserRole.get_db(), [r['id'] for r in roles]):
+    for role_doc in iter_docs(UserRole.get_db(), get_all_role_ids()):
```
Comment (esoergel) [inline corehq/apps/users/migrations/0024_add_login_as_all_users.py:12]: I'm betting you could probably just make all these migrations noop at this point, same with the cleanup management commands.

**Review by esoergel — APPROVED** [review]

File: corehq/apps/users/migrations/0024_add_login_as_all_users.py (hunk 3 above)
Comment (snopoke) [reply corehq/apps/users/migrations/0024_add_login_as_all_users.py:12]: I did consider that but decided not to do it as part of this PR. I will likely have to do that as part of the SQL migration though.

**Pull request merged**